app/controllers/comments_controller.rb

```diff
@@ -65,13 +67,6 @@ def set_comment
       @comment = Comment.find(params[:id])
     end
 
-    def is_an_authorized_user
-      @photo = Photo.find(params.fetch(:comment).fetch(:photo_id))
-      if current_user != @photo.owner && @photo.owner.private? && !current_user.leaders.include?(@photo.owner)
-        redirect_back fallback_location: root_url, alert: "Not authorized"
-      end
-    end
-    
     # Only allow a list of trusted parameters through.
     def comment_params
       params.require(:comment).permit(:author_id, :photo_id, :body)
```

Comment:
Ensure :author_id is not manipulatable by removing it from comment_params and securely assigning current_user as the author within the controller action.